src/main/java/tc/oc/pgm/modules/InfoModule.java

```diff
@@ -117,6 +135,48 @@ public static InfoModule parse(MapModuleContext context, Logger logger, Document
           throw new InvalidXMLException("Contributor must have either a name or UUID", child);
         }
 
+        // FIXME low quality code, but it gets the job done
+        if (uuid != null) {
+          if (Contributor.isUUIDCached(uuid)) {
+            name = Contributor.getCachedName(uuid);
+          } else {
+            Bukkit.getScheduler()
+                .runTaskAsynchronously(
+                    PGM.get(),
+                    () -> {
+                      try {
+                        HttpURLConnection con =
+                            (HttpURLConnection)
+                                new URL("https://api.ashcon.app/mojang/v2/user/" + uuid.toString())
+                                    .openConnection();
+                        con.setRequestMethod("GET");
+                        con.setRequestProperty("Accept", "application/json");
```

Comment:
Need to add a timeout? Also would add a user-agent.

Reply:
What would you prefer that the user-agent be?